catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -355,9 +358,42 @@ private String getExceptionMessage(String responseType, String previousException
         return builder.toString();
     }
 
+    public void shutdown() {
+        getPeriodicDrain().shutdown();
+    }
+
+    public String getRootBackupDir() {
+        return rootBackupDir.getAbsolutePath();
+    }
+
+    /**
+     * Sets the root file system backup directory.
+     *
+     * @param dir absolute path for the root file system backup directory.
+     */
+    public void setRootBackupDir(String dir) {
+        if (StringUtils.isBlank(dir)) {
+            LOGGER.error("The root backup directory is blank.");
```

Comment:
There was a unit test that make sure setting the directory to null didn't cause an NPE. Then it tested that an exception was thrown when ` catalogBackupPlugin.process`. My take away was that an exception was no supposed to be thrown until `process` was called.